compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1442,7 +1450,8 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                     (xform_self_ty, xform_ret_ty) =
                         self.xform_self_ty(probe.item, impl_ty, impl_args);
                     xform_self_ty = ocx.normalize(cause, self.param_env, xform_self_ty);
-                    match ocx.sup(cause, self.param_env, xform_self_ty, self_ty) {
+                    match ocx.relate(cause, self.param_env, self.variance(), xform_self_ty, self_ty)
```

Comment:
that's cursed... I feel like we should use `ty::Covariant` for method calls and actually flip `xform_self_ty` and `self_ty` here :blush:

Reply:
but then we change the "expected" and "found" 🤔 of the type errors I wonder if that affects diagnostics

Reply:
probably, the question is whether it makes them worse :grin:

Reply:
Yeah I'll test it